src/Component/Settlement/Cli/SettlementGenerateCommand.php

```
@@ -0,0 +1,107 @@
+            if (null === $nextSettlementStartDate || null === $nextSettlementEndDate) {
+                continue;
+            }
+            foreach ($channels as $channel) {
+                $lastSettlement = $this->settlementRepository->findLastByVendorAndChannel($vendor, $channel);
+                if (
+                    null !== $lastSettlement
+                    && $lastSettlement->getEndDate() > $nextSettlementStartDate
+                ) {
+                    continue;
+                }
+
+                ['total' => $total, 'commissionTotal' => $commissionTotal] = $this->orderRepository->findForSettlementByVendorAndChannelAndDates(
+                    $vendor,
+                    $channel,
+                    $nextSettlementStartDate,
+                    $nextSettlementEndDate
+                );
+                $nextSettlement = $this->settlementFactory->createNewForVendorAndChannel(
+                    $vendor,
+                    $channel,
+                    (int) $total,
+                    (int) $commissionTotal,
+                    $nextSettlementStartDate,
+                    $nextSettlementEndDate
+                );
+                $this->settlementManager->persist($nextSettlement);
+                if (0 === ($persistCount % 50)) {
+                    $this->settlementManager->flush();
+                }
+                ++$persistCount;
+            }
+        }
+        $this->settlementManager->flush();
+
+        return Command::SUCCESS;
+    }
+
+    private function getSettlementDateRangeFromVendor(VendorInterface $vendor): array
+    {
+        /** @var AbstractSettlementPeriodResolverStrategy $settlementPeriodResolver */
+        foreach ($this->settlementPeriodResolvers as $settlementPeriodResolver) {
+            if ($settlementPeriodResolver->supports($vendor)) {
+                return $settlementPeriodResolver->resolve();
+            }
+        }
+
+        return [null, null];
```

We should throw an exception here

---

fixed + added test